addons/account/models/account_bank_statement_line.py

```diff
@@ -463,7 +463,7 @@ def _find_or_create_bank_account(self):
                 'partner_id': self.partner_id.id,
                 'journal_id': None,
             })
-        return bank_account.filtered(lambda x: x.company_id in (False, self.company_id))
+        return bank_account.filtered(lambda x: not x.company_id or x.company_id == self.company_id)
```

Comment:
```suggestion
        return bank_account.filtered(lambda x: x.company_id.id in (False, self.company_id.id))
```
In those cases, I usually find it clearer to compare ids directly, instead of recordsets, what do you think ?